hw04-generics/src/main/java/homework/CustomerService.java

```diff
@@ -0,0 +1,36 @@
+package homework;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class CustomerService {
+    private final Comparator<Customer> byScore = Comparator.comparing(Customer::getScores);
+    private final TreeMap<Customer, String> map = new TreeMap<>(byScore);
```

Comment:
Тип лучше объявлять минимально необходимым интерфейсом:
```suggestion
    private final NavigableMap<Customer, String> map = new TreeMap<>(byScore);
```